### PR TITLE
maximumVolumeSize should be updated when it changes

### PR DESCRIPTION
**Pull request by mrpre**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change

 /kind bug

> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
`maximumVolumeSize` should be updated but not. 
**Which issue(s) this PR fixes**:
NONE

**Special notes for your reviewer**:
NONE
**Does this PR introduce a user-facing change?**:
NONE
```release-note
bug fix, no release-note
```

### CODE REVIEW

**Comment by pohly** [comment]
Good catch. Thanks for sharing. Can you also sign the CLA?

/ok-to-test

**Comment by mrpre** [comment]
CLA  signed and all tests passed @pohly @humblec @xing-yang

**Comment by humblec** [comment]
Thanks @mrpre .. lgtm..

File: pkg/capacity/capacity.go [hunk 1]
```diff
@@ -642,8 +642,10 @@ func (c *Controller) syncCapacity(ctx context.Context, item workItem) error {
 		// scenario that we end up creating two objects for the same work item, the second
 		// one will be recognized as duplicate and get deleted again once we receive it.
 	} else if capacity.Capacity.Value() == quantity.Value() &&
+		capacity.MaximumVolumeSize == maximumVolumeSize &&
+		(maximumVolumeSize == nil || capacity.MaximumVolumeSize.Value() == maximumVolumeSize.Value()) &&
```
Comment (pohly) [inline pkg/capacity/capacity.go:646]: This gets hard to review for correctness.

Let's move this into a helper function:
```
func sizesAreEqual(a, b *resource.Quantity) bool {
   if a == b {
     // Both nil or the same value.
     return true
  }
  if a == nil {
     // b must be non-nil.
     return false
  }
  // Both not nil, compare values.
  return a.Value() == b.Value()
}
```

**Review by pohly — APPROVED** [review]
/lgtm
/assign @jsafrane 

For approval.

**Pull request merged**